src/rtc/core.py

```diff
@@ -152,6 +156,327 @@ def save_as_cog(filename, scratch_dir = '.', logger = None,
+    transformation = osr.CoordinateTransformation(tile_srs, polygon_srs)
+
+    elevation = 0
+    tile_x_array = np.zeros((4))
+    tile_y_array = np.zeros((4))
+    tile_x_array[0], tile_y_array[0], z = transformation.TransformPoint(
+        tile_min_x_projected, tile_max_y_projected, elevation)
+    tile_x_array[1], tile_y_array[1], z = transformation.TransformPoint(
+        tile_max_x_projected, tile_max_y_projected, elevation)
+    tile_x_array[2], tile_y_array[2], z = transformation.TransformPoint(
+        tile_max_x_projected, tile_min_y_projected, elevation)
+    tile_x_array[3], tile_y_array[3], z = transformation.TransformPoint(
+        tile_min_x_projected, tile_min_y_projected, elevation)
+    tile_min_y = np.min(tile_y_array)
+    tile_max_y = np.max(tile_y_array)
+    tile_min_x = np.min(tile_x_array)
+    tile_max_x = np.max(tile_x_array)
+
+    # handles antimeridian: tile_max_x around +180 and tile_min_x around -180
+    # add 360 to tile_min_x, so it becomes a little greater than +180
+    if tile_max_x > tile_min_x + 340:
+        tile_min_x, tile_max_x = tile_max_x, tile_min_x + 360
+
+    tile_ring = ogr.Geometry(ogr.wkbLinearRing)
+    tile_ring.AddPoint(tile_min_x, tile_max_y)
+    tile_ring.AddPoint(tile_max_x, tile_max_y)
+    tile_ring.AddPoint(tile_max_x, tile_min_y)
+    tile_ring.AddPoint(tile_min_x, tile_min_y)
+    tile_ring.AddPoint(tile_min_x, tile_max_y)
+    tile_polygon = ogr.Geometry(ogr.wkbPolygon)
+    tile_polygon.AddGeometry(tile_ring)
+    tile_polygon.AssignSpatialReference(polygon_srs)
+    return tile_polygon, tile_min_y, tile_max_y, tile_min_x, tile_max_x
+
+
+
+def _antimeridian_crossing_requires_special_handling(
+        file_srs, file_min_x, tile_min_x, tile_max_x):
+    '''
+    Check if ancillary input requires special handling due to
+    the antimeridian crossing
+
+    Parameters
+    ----------
+    file_srs: osr.SpatialReference
+        Ancillary file spatial reference system (SRS)
+    file_min_x: float
+        Ancillary file min longitude value in degrees
+    tile_min_x: float
+        Tile min longitude value in degrees
+    tile_max_x: float
+        Tile max longitude value in degrees
+
+    Returns
+    -------
+    flag_requires_special_handling : bool
+        Flag that indicate if the ancillary input requires special handling
+    '''
+
+    # Flag to indicate if the if the tile crosses the antimeridian.
+    flag_tile_crosses_antimeridian = tile_min_x < 180 and tile_max_x >= 180
+
+    # Flag to test if the ancillary input file is in geographic
+    # coordinates and if its longitude domain is represented
+    # within the [-180, +180] range, rather than the [0, +360] interval.
+    # This is verified by the test `min_x < -170`. There's no specific reason
+    # why -170 is used. It could be -160, or even 0.
+    flag_input_geographic_and_longitude_not_0_360 = \
+        file_srs.IsGeographic() and file_min_x < -170
```

Comment:
```suggestion
    flag_input_geographic_and_longitude_not_0_360 = \
        file_srs.IsGeographic() and file_min_x < 0.0
```
If I understand this line correctly, `flag_input_geographic_and_longitude_not_0_360 ` will be`True` when `file_min_x` is out of the range [0, 360].
Let's assume some edge case that `file_min_x = -1.0`. then `flag_input_geographic_and_longitude_not_0_360` will be `False` because `file_min_x < -170` is `False`.
Please check if that is what you intended.

Reply:
Thank you, @seongsujeong . I prefer to use a number closer to `-180`, but not too close. The idea is that if the DEM domain is between, for example, [-160, 200] that would not cause a problem. The most common case is [0, 360], that's why I was using that range in the comments and variable name. But I updated the code and comments to make it clearer:

```
    # Flag to test if the ancillary input file is in geographic
    # coordinates and if its longitude domain is represented
    # within the [-180, +180] range, rather than, for example, inside
    # the [0, +360] interval.
    # This is verified by the test `min_x < -165`. There's no specific reason
    # why -165 is used. It could be -160, or even 0. However, testing for
    # -165 is more general than -160 or 0, but still not too close to -180.
    flag_input_geographic_and_longitude_lt_m165 = \
        file_srs.IsGeographic() and file_min_x < -165
```